snapshot/git/gitfiler/setup.go

```diff
@@ -4,49 +4,52 @@ import (
 	"io/ioutil"
 	"path"
 
+	"github.com/scootdev/scoot/common/stats"
 	"github.com/scootdev/scoot/os/temp"
 	"github.com/scootdev/scoot/snapshot/git/repo"
 )
 
 // Utilities for creating Checkouters.
 
-// RepoIniter lets a client initialize a Repository to use as a Reference Repository.
+// RepoIniter is an interface for initializing repositories
 type RepoIniter interface {
-	// Init initializes the Repository to use as a reference repository.
-	Init() (*repo.Repository, error)
+	// Init initializes the Repository to use as a reference repository
+	// Takes a StatsReceiver to support instrumenting repo initialization
+	Init(stat stats.StatsReceiver) (*repo.Repository, error)
 }
 
 // RepoIniter implementation
 type ConstantIniter struct {
 	Repo *repo.Repository
 }
 
-func (g *ConstantIniter) Init() (*repo.Repository, error) {
+func (g *ConstantIniter) Init(stat stats.StatsReceiver) (*repo.Repository, error) {
 	return g.Repo, nil
 }
 
-// A Pool that will only have a single repo, populated by repoIniter, that serves until doneCh is closed
-func NewSingleRepoPool(repoIniter RepoIniter, doneCh <-chan struct{}) *RepoPool {
-	singlePool := NewRepoPool(nil, nil, doneCh, 1)
-	go func() {
-		r, err := repoIniter.Init()
-		singlePool.Release(r, err)
-	}()
+// Util for creating a Pool that will only have a single repo
+func NewSingleRepoPool(repoIniter RepoIniter,
+	stat stats.StatsReceiver,
+	doneCh <-chan struct{}) *RepoPool {
+	singlePool := NewRepoPool(repoIniter, stat, []*repo.Repository{}, doneCh, 1)
```

Comment:
The previous implementation of a pool with only a single resource was a hack - the initer and getter funcs were nil-ed out so the pool could not initialize the resources, and then this manually initialized it.  With the capacity feature, we just give the pool a cap of 1 and it will initialize a single repo on its own.